curriculum/test/test-challenges.js

```diff
@@ -377,9 +374,9 @@ function populateTestsForLang({ lang, challenges, meta, superBlocks }) {
               );
             });
 
-            const { challengeType } = challenge;
+            const { challengeType, challengeFiles } = challenge;
 
-            if (hasNoSolution(challengeType)) return;
+            if (!challengeFiles) return;
```

Comment:
No, because `solution` exists for stupid reasons. E.g.
> `solution: [{ content: "This project has no solution"}]`

As I say:
> The challengeFiles === null condition is true for all USED challenge types in the hasNoSolution list.

Technically, it is also undefined. But, for the existing challenge types in `hasNoSolution` it is never `[]`.

Reply:
The issue is that we're using `!challengeFiles` as a proxy for "should not be tested". It's two steps removed from that (i.e. it's not equivalent to "does not have tests").

While it does make sense to rely less on `challengeType`, I don't think we should rely on a loose proxy instead.

How about we add a `hasTests` flag? Then we can check that instead.

> solution: [{ content: "This project has no solution"}]

I can try and clean that stuff up separately.

Reply:
`hasTests` does not work, because challenges with tests do not necessarily have solutions.